include/swift/AST/Module.h

```diff
@@ -967,6 +967,10 @@ class ModuleDecl
   /// The order of the results is not guaranteed to be meaningful.
   void getTopLevelDecls(SmallVectorImpl<Decl*> &Results) const;
 
+  /// Finds all top-level decls of this module including auxiliary decls.
+  void
+  getTopLevelDeclsWithAuxiliaryDecls(SmallVectorImpl<Decl *> &Results) const;
```

Comment:
This is currently a waste be cause non-SourFile FileUnit won't have macro expansions.

But in general,  I feel like this should be use more widely for example `ModuleDepGraphFactory::addAllDefinedDecls` currently use `getTopLevelDecls()` which doesn't contain macro expanded decls.

Reply:
But this will not regress the performance because, at this point, only `emitSwiftInterface()` uses this, and previously in `PrintAST::visit(Decl *)`, trying to visit auxiliary decls happened unconditionally for every decls including serialized.